SampleViewer/mainSample.cpp

```diff
@@ -45,252 +46,244 @@
-#include "OfflineRouting.h"
-#include "OpenMapUrl.h"
-#include "OpenMobileMap_MapPackage.h"
-#include "OpenMobileScenePackage.h"
-#include "OpenScene.h"
-#include "OrbitCameraAroundObject.h"
-#include "OSM_Layer.h"
-#include "PerformValveIsolationTrace.h"
-#include "Picture_Marker_Symbol.h"
-#include "PlayAKmlTour.h"
-#include "PortalUserInfo.h"
-#include "ProjectGeometry.h"
-#include "QueryFeaturesWithArcadeExpression.h"
-#include "QueryMapImageSublayer.h"
-#include "QueryOGCAPICQLFilters.h"
-#include "RasterColormapRenderer.h"
-#include "RasterFunctionService.h"
-#include "RasterLayerFile.h"
-#include "RasterLayerGeoPackage.h"
-#include "RasterLayerService.h"
-#include "RasterRenderingRule.h"
-#include "RasterRgbRenderer.h"
-#include "RasterStretchRenderer.h"
-#include "ReadGeoPackage.h"
-#include "ReadSymbolsFromMobileStyle.h"
-#include "RealisticLightingAndShadows.h"
-#include "ReverseGeocodeOnline.h"
-#include "RouteAroundBarriers.h"
-#include "SceneLayerSelection.h"
-#include "ScenePropertiesExpressions.h"
-#include "SearchDictionarySymbolStyle.h"
-#include "SearchForWebmap.h"
-#include "ServiceArea.h"
-#include "ServiceFeatureTableCache.h"
-#include "ServiceFeatureTableManualCache.h"
-#include "ServiceFeatureTableNoCache.h"
-#include "SetInitialMapArea.h"
-#include "SetInitialMapLocation.h"
-#include "SetMapSpatialReference.h"
-#include "SetMaxExtent.h"
-#include "ShowCallout.h"
-#include "ShowDeviceLocationUsingIndoorPositioning.h"
-#include "ShowLabelsOnLayers.h"
-#include "ShowLocationHistory.h"
-#include "ShowMagnifier.h"
-#include "ShowOrgBasemaps.h"
-#include "ShowPopup.h"
-#include "Simple_Marker_Symbol.h"
-#include "Simple_Renderer.h"
-#include "SketchOnMap.h"
-#include "SnapGeometryEdits.h"
-#include "SpatialOperations.h"
-#include "SpatialRelationships.h"
-#include "StatisticalQuery.h"
-#include "StatisticalQueryGroupSort.h"
-#include "StyleWmsLayer.h"
-#include "SuggestListModel.h"
-#include "SetSurfacePlacementMode.h"
-#include "SymbolizeShapefile.h"
-#include "Symbols.h"
-#include "SyncMapViewSceneView.h"
-#include "TakeScreenshot.h"
-#include "TerrainExaggeration.h"
-#include "TileCacheLayer.h"
-#include "TokenAuthentication.h"
-#include "TraceUtilityNetwork.h"
-#include "Unique_Value_Renderer.h"
-#include "UpdateAttributesFeatureService.h"
-#include "UpdateGeometryFeatureService.h"
-#include "ValidateUtilityNetworkTopology.h"
-#include "VectorTiledLayerUrl.h"
-#include "ViewContentBeneathTerrainSurface.h"
-#include "ViewPointCloudDataOffline.h"
-#include "ViewshedCamera.h"
-#include "ViewshedGeoElement.h"
-#include "ViewshedLocation.h"
-#include "Web_Tiled_Layer.h"
-#include "WmsLayerUrl.h"
-#include "WMTS_Layer.h"
+#include "../CppSamples/Analysis/AnalyzeHotspots/AnalyzeHotspots.h"
+#include "../CppSamples/Analysis/AnalyzeViewshed/AnalyzeViewshed.h"
+#include "../CppSamples/Analysis/DistanceMeasurementAnalysis/DistanceMeasurementAnalysis.h"
+#include "../CppSamples/Analysis/Geotriggers/Geotriggers.h"
```

Comment:
This entire block (and below) are where the fix is. Since we no longer add each sample's path to the compiler include path, we need to reference each one by relative path now.

Reply:
Just an idea but could we create a separate header, say `CppSamplesRelativePaths.h`, and put this block of sample header includes there?

I did see for a similar idea you mentioned that it's just one more place to remember which i agree. It just gets under my skin that we have wall of header includes for the samples and we also have a wall of `::init` for each sample. Between those two, it makes up about 350 lines of 635 lines of this file.

Reply:
It's a good point, but I think in this case it has to go somewhere, so we're not saving much by putting it in a separate file. We know those headers are needed in this file, but if someone is perusing the repo, it will not be clear what the purpose of `CppSamplesRelativePaths.h` is by itself. So I lean towards leaving them here.